control-plane/connect-inject/peering_acceptor_controller.go

```diff
@@ -52,16 +54,32 @@ func (r *PeeringAcceptorController) Reconcile(ctx context.Context, req ctrl.Requ
 	// If the PeeringAcceptor resource has been deleted (and we get an IsNotFound
 	// error), we need to delete it in Consul.
 	if k8serrors.IsNotFound(err) {
-		r.Log.Info("PeeringAcceptor was deleted, deleting from Consul", "name", req.Name, "ns", req.Namespace)
-		if err := r.deletePeering(ctx, req.Name); err != nil {
-			return ctrl.Result{}, err
-		}
+		r.Log.Info("PeeringAcceptor resource not found. Ignoring.", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, nil
 	} else if err != nil {
 		r.Log.Error(err, "failed to get PeeringAcceptor", "name", req.Name, "ns", req.Namespace)
 		return ctrl.Result{}, err
 	}
 
+	if acceptor.GetDeletionTimestamp().IsZero() {
+		if !controllerutil.ContainsFinalizer(acceptor, FinalizerName) {
+			controllerutil.AddFinalizer(acceptor, FinalizerName)
+			if err := r.Update(ctx, acceptor); err != nil {
+				return ctrl.Result{}, err
+			}
+		}
+	} else {
+		if containsString(acceptor.Finalizers, FinalizerName) {
+			r.Log.Info("PeeringAcceptor was deleted, deleting from Consul", "name", req.Name, "ns", req.Namespace)
+			if err := r.deletePeering(ctx, req.Name); err != nil {
```

Comment:
Could we also delete the generated K8s secret specified in the status when the PeeringAcceptor is deleted?

Reply:
I think it would be reasonable to do for the Acceptor but not the Dialer, since the Acceptor is the one that generates that secret, vs on the Dialer the secret isn't managed by the controller.

Reply:
great catch!!